### PR TITLE
Add API test to the test automation engineer tech test

### PR DESCRIPTION
**Pull request by smarkey**

### CODE REVIEW

File: Test.Automation.Engineer.md [hunk 1]
```diff
@@ -37,27 +42,90 @@ Scenario: Search for restaurants in an area
...
+    },
+  }
+}
+```
+
+2. Verify the following functional requirements regarding restaurants data in response:
+
+   - `A restaurant with more than 1 rating should have a star rating greater than 0`
+   - `A restaurant with no ratings should have a star rating of 0`
+   - `A restaurant should have a valid Url (returns 2**)`
```
Comment (smarkey) [inline Test.Automation.Engineer.md]: Any reason to leave this open to 2xx? A GET to the url will always be 200

File: Test.Automation.Engineer.md [hunk 2]
```diff
@@ -37,27 +42,90 @@ Scenario: Search for restaurants in an area
...
+- Endpoint: `/restaurants/bypostcode/{postcode}`
+- Method: `GET`
+
+#### Example
+
+`GET` <https://uk.api.just-eat.io/restaurants/bypostcode/bs395rp> returns complex restaurants-related data for postcode `BS39 5RP`
+
+#### Tasks
+
+1. Verify `Restaurants[n].Address` field of each restaurant object in response. It should reflect the following schema:
```
Comment (smarkey) [inline Test.Automation.Engineer.md]: Change to `Verify Restaurants[n].Address object` rather than field

File: Test.Automation.Engineer.md [hunk 3]
```diff
@@ -37,27 +42,90 @@ Scenario: Search for restaurants in an area
...
-  1. What do you think is the most interesting trend in test automation?
-  1. How would you implement test automation in a legacy application? Have you ever had to do this?
-  1. How would you improve the customer experience of the Just Eat website?
-  1. Please describe yourself using JSON.
-
+1. How long did you spend on the technical test? What would you add to your solution if you had more time? If you didn't spend much time on the technical test then use this as an opportunity to explain what you would add.
+1. What do you think is the most interesting trend in test automation?
+1. How would you implement test automation in a legacy application? Have you ever had to do this?
+1. How would you improve the customer experience of the Just Eat website?
+1. Please describe yourself using JSON.
```
Comment (smarkey) [inline Test.Automation.Engineer.md]: Does this last task become superfluous and a drain on time now that they've already proven they understand JSON?

File: Test.Automation.Engineer.md [hunk 4]
```diff
@@ -37,27 +42,90 @@ Scenario: Search for restaurants in an area
...
 
 Please answer the following questions in a markdown file called `Answers to technical questions.md`.
 
-  1. How long did you spend on the technical test? What would you add to your solution if you had more time? If you didn't spend much time on the technical test then use this as an opportunity to explain what you would add.
-  1. What do you think is the most interesting trend in test automation?
-  1. How would you implement test automation in a legacy application? Have you ever had to do this?
-  1. How would you improve the customer experience of the Just Eat website?
-  1. Please describe yourself using JSON.
-
+1. How long did you spend on the technical test? What would you add to your solution if you had more time? If you didn't spend much time on the technical test then use this as an opportunity to explain what you would add.
```
Comment (smarkey) [inline Test.Automation.Engineer.md]: Should we split this into three numbered points? Seems needless to group them into one harder-to-read point

File: Test.Automation.Engineer.md (hunk 3 above)
Comment (eignatyev) [reply Test.Automation.Engineer.md]: :D removing

File: Test.Automation.Engineer.md [hunk 5]
```diff
@@ -37,27 +42,90 @@ Scenario: Search for restaurants in an area
...
+
+#### Prerequisites
+
+- URL: <https://uk.api.just-eat.io>
+- Endpoint: `/restaurants/bypostcode/{postcode}`
+- Method: `GET`
+
+#### Example
+
+`GET` <https://uk.api.just-eat.io/restaurants/bypostcode/bs395rp> returns complex restaurants-related data for postcode `BS39 5RP`
```
[the remark this answers is not in the document]
Comment (eignatyev) [reply Test.Automation.Engineer.md]: It's just an example. I'm personally interested in seeing how candidates choose a postcode themselves.
Happy to put AR51 1AA here

File: Test.Automation.Engineer.md (hunk 2 above)
Comment (smarkey) [reply Test.Automation.Engineer.md]: Sub-object is perfect!

File: Test.Automation.Engineer.md [hunk 6]
```diff
@@ -37,27 +42,91 @@ Scenario: Search for restaurants in an area
...
+    },
+  }
+}
+```
+
+2. Verify the following functional requirements regarding restaurants data in response:
+
+   - `A restaurant with more than 1 rating should have a star rating greater than 0`
+   - `A restaurant with no ratings should have a star rating of 0`
+   - `A restaurant should have a valid Url (returns 200)`
```
Comment (martincostello) [inline Test.Automation.Engineer.md]:
```suggestion
   - `A restaurant should have a valid URL (returns 200)`
```

File: Test.Automation.Engineer.md [hunk 7]
```diff
@@ -37,27 +42,91 @@ Scenario: Search for restaurants in an area
...
+
+1. Design 3 more requirements which you think are missing most and implement verification tests.
+
+#### Platform Choice
+
+You can create the tests using any language or framework you are experienced with.
+Please write them programmatically as a script produced by a tool such as Postman or SoapUI is not sufficient.
```
Comment (martincostello) [inline Test.Automation.Engineer.md]:
```suggestion
Please write them programmatically; a script produced by a tool such as Postman or SoapUI is not sufficient.
```

**Pull request merged**